src/RpcClient.ts

```diff
@@ -354,4 +378,94 @@ export class RpcClient {
+   * @throws {Error}
+   */
+  async mintCNFT(
+    request: MintCNFTRequest
+  ): Promise<MintCNFTResponse | undefined> {
+    try {
+      let ownerKeypair = loadWallet(request.ownerKeypair);
+      let defaultMetadataArgs: MetadataArgs = {
+        ...request.metadataArgs,
+        primarySaleHappened: false,
+        isMutable: false,
+        editionNonce: 0,
+        tokenStandard: null,
+        uses: null,
+        tokenProgramVersion: null,
+      };
+      let treeWallet: Keypair;
+      if (!request.treeKeypair) {
+        treeWallet = Keypair.generate();
+        await initTree(this.connection, ownerKeypair, treeWallet);
+        console.log("New Merkle Tree Private Key: ", treeWallet.secretKey);
+      } else {
+        treeWallet = loadWallet(request.treeKeypair);
+      }
+      let collectionMint: PublicKey;
+      if (!request.collectionMint) {
+        const {
+          collectionMint,
+          collectionMetadataAccount,
+          collectionMasterEditionAccount,
+        } = await initCollection(
+          this.connection,
+          ownerKeypair,
+          request.metadataArgs.name,
+          request.metadataArgs.symbol,
+          request.metadataArgs.uri
+        );
+        const signature = await mintCompressedNft(
+          this.connection,
+          defaultMetadataArgs,
+          request.ownerKeypair,
+          treeWallet,
+          collectionMint,
+          collectionMetadataAccount,
+          collectionMasterEditionAccount,
+        );
+        const response: MintCNFTResponse = {
+          creator: request.ownerKeypair,
+          collectionName: request.metadataArgs.name,
+          collectionSymbol: request.metadataArgs.symbol,
+          collectionUri: request.metadataArgs.uri,
+          collectionMint: collectionMint.toString(),
+          treeId: treeWallet.publicKey.toString(),
+          treeKeypair: treeWallet.secretKey,
+          signature: signature,
+        };
+        return response as MintCNFTResponse;
+      } else {
+        collectionMint = request.collectionMint;
+        const { collectionMetadataAccount, collectionMasterEditionAccount } =
+          await existingCollection(collectionMint);
+        const signature = await mintCompressedNft(
+          this.connection,
+          defaultMetadataArgs,
+          request.ownerKeypair,
+          treeWallet,
+          collectionMint,
+          collectionMetadataAccount,
+          collectionMasterEditionAccount,
+        );
+
+        const response: MintCNFTResponse = {
+          creator: request.ownerKeypair,
+          collectionName: request.metadataArgs.name,
+          collectionUri: request.metadataArgs.uri,
+          collectionSymbol: request.metadataArgs.symbol,
+          collectionMint: collectionMint.toString(),
+          treeId: treeWallet.publicKey.toString(),
+          treeKeypair: treeWallet.secretKey,
+          signature: signature,
+        };
+        return response as MintCNFTResponse;
```

Comment:
You should return the assetID, too.

Lastly, I think we should be polling the DAS endpoint to confirm it got indexed and returning the response for people. It'll be better UX that way. This can be an option though (enabled by default)